src/en/ss14-by-example/ui-and-you.md

```diff
@@ -0,0 +1,474 @@
+properties on the `ColorPalette` class. From brightest to darkest, the properties
+are (as of writing) arranged as so:
+
+-   `+0`: `Text` `Base`
+-   `-1`: `TextDark`, `Element`
+-   `-2`: `BackgroundLight`, `PressedElement`
+-   `-3`: `Background`
+-   `-4`: `BackgroundDark`, `DisabledElement`
+
+Although it would be simpler to have the palettes be arrays of colors, I noticed
+that colors were used in basically three ways: Text, foreground elements, and
+background elements. So I represented that in the palette itself! It just makes
+the code a bit more readable.
+
+All palettes are defined in `Palettes.cs` as static properties. You can read
+them from anywhere! Neat!
+
+#### `ISheetletConfig`
+
+`ISheetletConfig` is intended to cut down on repeated code by providing shared
+functionality and definitions between the stylesheets.
+Any `Sheetlet` that requires the values in some instance of
+`ISheetletConfig` should have a generic type constraint that requires the
+`ISheetletConfig` interface.
+
+```cs
+[CommonSheetlet] // don't forget `[CommonSheetlet]`!
+public sealed class ExampleSheetlet<T> : Sheetlet<T> where T : PalettedStylesheet, IExampleConfig
+```
+
+`ISheetletConfig` also serves as a dependency check. When the stylesheets pull
+all the sheetlets that have `[CommonSheetlet]`, they will first check that they
+satisfy the type constraint before adding the rules to the stylesheet.
+
+```admonish info
+A sheetlet failing to satisfy the type constraints of any stylesheet will cause
+the game to log an error. If your styles are not showing up, this may be why.
+```
+
+#### Resource Access
+
+You should access resources differently in sheetlets. Each stylesheet provides a
+list of resource roots to look in, when requesting a resource of a certain type.
+For example, the root for `NanotrasenStylesheet` for `TextureResource` is
+`/Textures/Interface/Nano`. Any texture requested with `GetTexture` will will
+append the provided relative path to the root, and return the texture if it exists.
+
+This system is built to work with any resource type, though currently only textures
+are used in the sheetlets.
+
+### Generic Sheetlets
+
+As stated earlier, generic sheetlets are used for generic UI elements that are
+used in many different UIs.
+
+-   You should always select elements with `.Class` and not `.Identifier`.
+-   When accessing resources, use the `GetTextureOr` method to get the texture and
+    provide a fallback root to use if the texture is not found within the stylesheets roots.
+-   Avoid manual hardcoding of classes. When referencing classes,
+    please only use classes defined on the element being styled (in `StyleClass*` properties)
+    or define your own in `StyleClass.cs`.
+-   Avoid manual hardcoding of colors. Use the palettes provided by the stylesheet
+    to set the colors of elements. There are very few cases where you will actually
+    need to manually hardcode a color in a generic sheetlet.
```

Comment:
```suggestion
-   Use the palettes provided by the set stylesheet instead of specifying 
    colors by hand. There are very few cases where you will actually need
    to manually hardcode a color in a generic sheetlet.
```